src/librbd/ExclusiveLock.cc

```diff
@@ -140,12 +140,14 @@ template <typename I>
 void ExclusiveLock<I>::shut_down(Context *on_shut_down) {
   ldout(m_image_ctx.cct, 10) << dendl;
 
+  get();
```

Comment:
Avoid using `get` / `put` explicitly and instead do:
```suggestion
  auto ref = ceph::ref_t(this);
```

Reply:
done.